Thanks @TheGarkine for the proposed implementation. As a matter of fact we already have a `non-interactive` mode for the pulumi command, and I'd rather use that instead of passing the `--yes` flag. 

So my thinking is to use a more descriptive `--non-interactive` / `-n` flag which attempts to execute commands for the underlying IaC provider in non interactive mode. 

I tested using: 
```
runPulumiCommand(command, false, this.twilioClient); 
```
and the resources are created correctly without any prompt to the user. The only thing is that I don't see any printout but that may be a bug in the implementation of the non interactive `else` branch (I need more investigations). 

Still, I really like your idea of passing down pulumi's flag into the CLI, so we should probably work on that separately in another PR. 

Back to this PR, can I ask you to make the following changes: 

* Fix the linter errors (I'm about to reinforce linting in this project): 
```
/plugin-twilio-infra/src/commands/infra/deploy.js
   3:27   error  Strings must use singlequote  quotes
  24:20   error  Strings must use singlequote  quotes
  40:106  error  Missing trailing comma        comma-dangle
  41:7    error  Missing trailing comma        comma-dangle

/plugin-twilio-infra/src/commands/infra/destroy.js
   3:27   error  Strings must use singlequote  quotes
  13:20   error  Strings must use singlequote  quotes
  29:107  error  Missing trailing comma        comma-dangle
  30:7    error  Missing trailing comma        comma-dangle
```
* Change `--yes` to `--non-interactive` / `-n`
* Change the call to `runPulumiCommand()` to: 
```
runPulumiCommand(command, !this.flags.nonInteractive, this.twilioClient);
```

Let me know what you think, and thanks for your contributions!!

--- a/src/commands/infra/deploy.js
+++ b/src/commands/infra/deploy.js
@@ -1,6 +1,6 @@
 const { TwilioClientCommand } = require('@twilio/cli-core').baseCommands;
 const { TwilioCliError } = require('@twilio/cli-core').services.error;
-const { flags } = require("@oclif/command");
+const { flags } = require('@oclif/command');
 
 const { addInfra } = require('../../infra');
 
@@ -19,11 +19,13 @@ class InfraDeploy extends TwilioClientCommand {
         `The current stack is already deployed to ${deployment}. Please switch to that profile or define a new environment`
       );
     }
+
     let command = ['up'];
-    if (this.flags.yes) {
+    if (this.flags['non-interactive']) {
       command.push("--yes");
     }
     runPulumiCommand(command, true, this.twilioClient);
+
     try {
       // Store account SID of the project used for deployment
       addInfra(this.twilioClient.accountSid, getPulumiStack(), true);
@@ -35,12 +37,12 @@ class InfraDeploy extends TwilioClientCommand {
 
 InfraDeploy.flags = Object.assign(
   {
-    yes: flags.boolean({
-      char: 'y',
-      description: 'Passes through the --yes option to pulumi. Deploys without interactive confirmation.'
+    'non-interactive': flags.boolean({
+      char: 'n',
+      description: 'Deploys without interactive confirmation.',
     })
   },
-  TwilioClientCommand.flags
+  TwilioClientCommand.flags,
 );
 
 InfraDeploy.description =

--- a/src/commands/infra/destroy.js
+++ b/src/commands/infra/destroy.js
@@ -1,6 +1,6 @@
 const { TwilioClientCommand } = require('@twilio/cli-core').baseCommands;
 const { TwilioCliError } = require('@twilio/cli-core').services.error;
-const { flags } = require("@oclif/command");
+const { flags } = require('@oclif/command');
 
 const { destroyInfra } = require('../../infra');
 const { runPulumiCommand, Printer } = require('../../utils');
@@ -8,11 +8,13 @@ const { runPulumiCommand, Printer } = require('../../utils');
 class InfraDestroy extends TwilioClientCommand {
   async run() {
     await super.run();
+
     let command = ['destroy'];
-    if (this.flags.yes) {
+    if (this.flags["non-interactive"]) {
       command.push("--yes");
     }
     await runPulumiCommand(command, true, this.twilioClient);
+
     try {
       destroyInfra(this.twilioClient.accountSid);
       Printer.printSuccess('Resource(s) destroyed successfully!');
@@ -24,12 +26,12 @@ class InfraDestroy extends TwilioClientCommand {
 
 InfraDestroy.flags = Object.assign(
   {
-    yes: flags.boolean({
-      char: 'y',
-      description: 'Passes through the --yes option to pulumi. Destroys without interactive confirmation.'
+    'non-interactive': flags.boolean({
+      char: 'n',
+      description: 'Destroys without interactive confirmation.',
     })
   },
-  TwilioClientCommand.flags
+  TwilioClientCommand.flags,
 );
 
 InfraDestroy.description =

